KennyExternalProvidersLab/app/src/main/java/com/korbkenny/kennyexternalproviderslab/CalendarRecyclerViewAdapter.java

```diff
@@ -0,0 +1,57 @@
+package com.korbkenny.kennyexternalproviderslab;
+
+import android.database.Cursor;
+import android.provider.BaseColumns;
+import android.provider.CalendarContract;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.List;
+
+/**
+ * Created by KorbBookProReturns on 11/28/16.
+ */
+
+public class CalendarRecyclerViewAdapter extends RecyclerView.Adapter<CalendarViewHolder> {
+    private List<Event> mEventList;
+
+    public CalendarRecyclerViewAdapter(List<Event> eventList) {
+        mEventList = eventList;
+    }
+
+    @Override
+    public CalendarViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        return new CalendarViewHolder(inflater.inflate(android.R.layout.simple_list_item_1,parent,false));
+    }
+
+    @Override
+    public void onBindViewHolder(CalendarViewHolder holder, int position) {
+        holder.mName.setText(mEventList.get(position).getName());
+        holder.mDate.setText(mEventList.get(position).getDate());
+    }
```

Comment:
The only think you're missing is the ability to delete items from the calendar. Set an `OnClickListener` or `OnLongClickListener` on the item, then when it fires, to delete the event, do
```java
long eventId = mEventList.get(holder.getAdapterPosition()).getId();
Uri uri = ContentUris.withAppendedId(CalendarContract.Events.CONTENT_URI, eventId);
holder.mName.getContext().getContentResolver().delete(uri, null, null);
```